### PR TITLE
Feature/3 reallocate and load

### PR DESCRIPTION
**Pull request by kevintumbo**
created tests and functionalty for Reallocate person and load file.

### CODE REVIEW

File: Tests/person_tests.py [hunk 1]
```diff
@@ -56,3 +56,124 @@ def test_cannot_add_person_with_missing_arguments(self):
...
+        msg1 = self.dojo.reallocate_person("James", "macavoy", "kenya")
+        self.assertEquals(msg1, "Sorry that person does not exist")
+
+    def test_cannot_reallocate_person_if_room_does_not_exist(self):
+        """ Test failure when reallocating person to room that does not exist """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.add_person('chris', 'Jaba', 'FELLOW', 'Y')
+        self.dojo.create_room("office", ["Somalia"])
```
Comment (oscamaina) [inline Tests/person_tests.py:143]: line 138 is unnecessary at this point.

File: Tests/person_tests.py [hunk 2]
```diff
@@ -56,3 +56,124 @@ def test_cannot_add_person_with_missing_arguments(self):
         self.dojo.add_person('', '', '', '')
         self.assertEqual(len(self.dojo.total_people), self.num_of_people,
                          'Person(s) cannot be created without necessary credentials')
+
+    def test_successful_loading_of_people_using_from_a_txt_file(self):
```
Comment (oscamaina) [inline Tests/person_tests.py]: method name, i suggest you change it to communicate in a proper way

File: Tests/person_tests.py [hunk 3]
```diff
@@ -56,3 +56,124 @@ def test_cannot_add_person_with_missing_arguments(self):
...
+        self.assertEqual(len(self.dojo.total_people), 0)
+        self.assertEquals(load, "Sorry that file does not exist")
+
+    def test_unsuccessful_loading_of_people_due_to_wrong_format_in_text_file(self):
+        """ Test for unsuccessful addition of person(s) due to wrong format in text file """
+
+        self.dojo = Dojo()
+        self.dojo.load_people("wrong_format")
+        self.assertEqual(len(self.dojo.total_people), 0)
+        self.assertEqual(len(self.dojo.total_people), 0)
```
Comment (oscamaina) [inline Tests/person_tests.py]: DRY,line 80 and 81 are the same.

File: Tests/person_tests.py [hunk 4]
```diff
@@ -56,3 +56,124 @@ def test_cannot_add_person_with_missing_arguments(self):
...
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.add_person('Ian', 'pyr', 'FELLOW')
+        self.dojo.add_person('Joe', 'jameson', 'FELLOW')
+        self.dojo.add_person('Ruben', 'konia', 'FELLOW')
+        self.dojo.add_person('James', 'macavoy', 'FELLOW')
+        self.dojo.add_person('Akira', 'Menai', 'FELLOW')
+        self.dojo.add_person('rex', 'acey', 'FELLOW')
+        self.assertTrue(len(self.dojo.total_rooms[0].occupants) == 6)
+        self.dojo.add_person('Larry', 'smith', 'FELLOW')
```
Comment (mnjaggah) [inline Tests/person_tests.py:159]: this person is not allocated to any room yet to be reallocated

File: Tests/person_tests.py [hunk 5]
```diff
@@ -56,3 +56,124 @@ def test_cannot_add_person_with_missing_arguments(self):
...
+
+    def test_cannot_reallocate_person_if_room_does_not_exist(self):
+        """ Test failure when reallocating person to room that does not exist """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.add_person('chris', 'Jaba', 'FELLOW', 'Y')
+        self.dojo.create_room("office", ["Somalia"])
+        msg1 = self.dojo.reallocate_person("chris", "Jaba", "Ghana")
+        self.assertEquals(msg1, "Sorry that person does not exist")
```
Comment (mnjaggah) [inline Tests/person_tests.py]: *room does not exist

File: Tests/person_tests.py (hunk 1 above)
Comment (kevintumbo) [reply Tests/person_tests.py:143]: got it.

File: Tests/person_tests.py (hunk 2 above)
Comment (kevintumbo) [reply Tests/person_tests.py]: ok. will do.

File: Tests/person_tests.py (hunk 3 above)
Comment (kevintumbo) [reply Tests/person_tests.py]: will change

File: Tests/person_tests.py (hunk 4 above)
Comment (kevintumbo) [reply Tests/person_tests.py:159]: In the functionality, i check whether a person is waiting for allocation. if so  he is allocated a room

File: Tests/person_tests.py (hunk 5 above)
Comment (kevintumbo) [reply Tests/person_tests.py]: thanks for spotting that.

File: Tests/person_tests.py (hunk 4 above)
Comment (mnjaggah) [reply Tests/person_tests.py:159]: oh, thats okay then

File: dojo.py [hunk 6]
```diff
@@ -218,3 +219,78 @@ def print_unallocated(self, filename=None):
...
+
+        if len(room_to_relocate.occupants) == room_to_relocate.maximum_occupants:
+            return "sorry that room is full"
+
+        else:
+            if room_to_relocate.type == "office":
+                if person_reallocating in self.waiting_for_office_allocation:
+                    self.waiting_for_office_allocation.remove(person_reallocating)
+                    room_to_relocate.occupants.append(person_reallocating)
+                    return "You has been removed from office waiting list to {0} {1}" \
```
Comment (GDP254) [inline dojo.py]: This sentence is a bit odd in my opinion. If it started with the person in the office list this would help.

File: dojo.py [hunk 7]
```diff
@@ -218,3 +219,78 @@ def print_unallocated(self, filename=None):
                 file.write("\n" + "------------------------------------------\n")
                 file.write("{}".format(str1) + "\n")
                 file.close()
+
+    def reallocate_person(self, first_name, last_name, new_room_name):
```
Comment (GDP254) [inline dojo.py:230]: This function is really long in my opinion. However, I find that the spacing and naming make it easy enough to follow.

Comment (kevintumbo) [reply dojo.py:230]: Yeah the length also bothered me... I'll work on making it shorter.

**Pull request closed without merging**